proto/kuksa/val/v1/types.proto

```diff
@@ -11,278 +11,288 @@
-  DATA_TYPE_TIMESTAMP_ARRAY = 32;
-}
-
-// Entry type
-enum EntryType {
-  ENTRY_TYPE_UNSPECIFIED = 0;
-  ENTRY_TYPE_ATTRIBUTE   = 1;
-  ENTRY_TYPE_SENSOR      = 2;
-  ENTRY_TYPE_ACTUATOR    = 3;
-}
-
-// A `View` specifies a set of fields which should
-// be populated in a `DataEntry` (in a response message)
-enum View {
-  VIEW_UNSPECIFIED   = 0;   // Unspecified. Equivalent to VIEW_CURRENT_VALUE unless `fields` are explicitly set.
-  VIEW_CURRENT_VALUE = 1;   // Populate DataEntry with value.
-  VIEW_TARGET_VALUE  = 2;   // Populate DataEntry with actuator target.
-  VIEW_METADATA      = 3;   // Populate DataEntry with metadata.
-  VIEW_FIELDS        = 10;  // Populate DataEntry only with requested fields.
-  VIEW_ALL           = 20;  // Populate DataEntry with everything.
-}
-
-// A `Field` corresponds to a specific field of a `DataEntry`.
-//
-// It can be used to:
-//   * populate only specific fields of a `DataEntry` response.
-//   * specify which fields of a `DataEntry` should be set as
-//     part of a `Set` request.
-//   * subscribe to only specific fields of a data entry.
-//   * convey which fields of an updated `DataEntry` have changed.
-enum Field {
-  FIELD_UNSPECIFIED                = 0;   // "*" i.e. everything
-  FIELD_PATH                       = 1;   // path
-  FIELD_VALUE                      = 2;   // value
-  FIELD_ACTUATOR_TARGET            = 3;   // actuator_target
-  FIELD_METADATA                   = 10;  // metadata.*
-  FIELD_METADATA_DATA_TYPE         = 11;  // metadata.data_type
-  FIELD_METADATA_DESCRIPTION       = 12;  // metadata.description
-  FIELD_METADATA_ENTRY_TYPE        = 13;  // metadata.entry_type
-  FIELD_METADATA_COMMENT           = 14;  // metadata.comment
-  FIELD_METADATA_DEPRECATION       = 15;  // metadata.deprecation
-  FIELD_METADATA_UNIT              = 16;  // metadata.unit
-  FIELD_METADATA_VALUE_RESTRICTION = 17;  // metadata.value_restriction.*
-  FIELD_METADATA_ACTUATOR          = 20;  // metadata.actuator.*
-  FIELD_METADATA_SENSOR            = 30;  // metadata.sensor.*
-  FIELD_METADATA_ATTRIBUTE         = 40;  // metadata.attribute.*
-}
-
-// Error response shall be an HTTP-like code.
-// Should follow https://www.w3.org/TR/viss2-transport/#status-codes.
-message Error {
-  uint32 code    = 1;
-  string reason  = 2;
-  string message = 3;
-}
-
-// Used in get/set requests to report errors for specific entries
-message DataEntryError {
-  string path = 1;  // vss path
-  Error error = 2;
-}
-
-message StringArray {
-  repeated string values = 1;
-}
-
-message BoolArray {
-  repeated bool values = 1;
-}
-
-message Int32Array {
-  repeated sint32 values = 1;
-}
-
-message Int64Array {
-  repeated sint64 values = 1;
-}
-
-message Uint32Array {
-  repeated uint32 values = 1;
-}
-
-message Uint64Array {
-  repeated uint64 values = 1;
-}
-
-message FloatArray {
-  repeated float values = 1;
-}
-
-message DoubleArray {
-  repeated double values = 1;
-}
+ syntax = "proto3";
```

Comment:
Changing indentation to 1 for everything? Any special reason?

Reply:
I have pushed a cleaner commit, it seems my proto editor does not work